bears/installation_tool.py

```diff
@@ -0,0 +1,161 @@
+from coalib.collecting.Importers import iimport_objects
+from coalib.misc.Shell import call_without_output
+from prompt_toolkit import prompt
+from prompt_toolkit.contrib.completers import WordCompleter
+from pygments.lexers import SqlLexer
+from pyprint.ConsolePrinter import ConsolePrinter
+
+
+def install_pip_package(package_name):
+    """
+    Uses call to install a PyPi package.
+
+    :param package_name: The package to be installed.
+    """
+    print('To install the dependencies correctly, we will need administrator '
+          'rights from you.')
+    subprocess.call(['sudo', sys.executable, '-m',
+                     'pip', 'install', package_name, '--upgrade'])
+
+
+def get_output(command):
+    """
+    Runs the command and decodes the output and returns it.
+
+    :param command: The command to be run.
+    :return:        The output of the command, decoded.
+    """
+    function = subprocess.Popen(command, stdout=subprocess.PIPE)
+    return function.communicate()[0].decode("utf-8")
+
+
+def check_failed_bears(bear_list):
+    """
+    Checks if any bear failed installation. Exits with 1 if so.
+
+    :param bear_list: List of the failed bears.
+    """
+    if bear_list:
+        print('Bears that failed installing:' + "\n".join(bear_list))
+        sys.exit(1)
+
+
+def install_requirements(package_name):
+    """
+    Imports a package and tries installing its requirements.
+
+    :param package_name:        The package to be imported.
+    :param package_failed_list: The list with the packages which their
+                                requirements failed installing.
+    :return:                    A list with the packages which had their
+                                requirements failing to be installed.
+    """
+    package_failed_list = []
+    package_object = importlib.import_module(
+        package_name + "." + package_name)
+    for requirement in getattr(package_object, package_name).REQUIREMENTS:
+        try:
+            subprocess.call([requirement.install_command()])
+        except:
+            package_failed_list.append(package_name)
+    return package_failed_list
+
+
+def get_all_bears_names_from_PyPI():
+    """
+    Gets all the bears names from PyPI, using the link in the description.
+
+    :return: A list with all the bear names.
+    """
+    output = get_output(['pip', 'search', "coala.rtfd.org"])
+    return re.findall(r'(.+) \(\d.*\n', output)
+
+
+def install_bears(bear_names_list, install_deps):
+    """
+    Tries to install each bear from the bear_names_list. Will also check for
+    bears which failed to be installed, or their requirements failed to be
+    installed.
+
+    :param bear_names_list: The list which contains the names of the bears.
+    :param install_deps:    An arg which is given to also install the bears'
+                            dependencies.
+    """
+    for bear_name in bear_names_list:
+        bear_name = bear_name.replace(' ', '')
```

Comment:
Where would this be needed? Wouldn't the bears already be one word with no space?